src/main/java/com/avispl/symphony/dal/avdevices/camera/panasonic/awue150/CameraPanasonicAWUE150Communicator.java

```diff
@@ -0,0 +1,1779 @@
+
+	/**
+	 * This method is used to populate focus control
+	 *
+	 * @param stats store all statistics
+	 * @param advancedControllableProperties store all controllable properties
+	 */
+	private void populatePresetControls(Map<String, String> stats, List<AdvancedControllableProperty> advancedControllableProperties) {
+		String groupName = DevicesMetricGroup.PRESET_CONTROL.getName() + DeviceConstant.HASH;
+		if (presetModes.isEmpty()) {
+			presetModes.add(DeviceConstant.DEFAULT_PRESET);
+			for (int index = DeviceConstant.PRESET_INDEX_MIN; index <= DeviceConstant.PRESET_INDEX_MAX; index++) {
+				presetModes.add(PresetControlMetric.PRESET.getName().concat(String.format(DeviceConstant.THREE_NUMBER_FORMAT, index)));
+			}
+		}
+		addAdvanceControlProperties(advancedControllableProperties, createDropdown(stats, groupName.concat(PresetControlMetric.PRESET.getName()), presetModes, cachedPresetControl));
+
+		addAdvanceControlProperties(advancedControllableProperties, createButton(stats, groupName.concat(PresetControlMetric.HOME.getName()), DeviceConstant.HOME, DeviceConstant.PUSHING));
+		if (!DeviceConstant.DEFAULT_PRESET.equals(cachedPresetControl)) {
+			addAdvanceControlProperties(advancedControllableProperties, createButton(stats, groupName.concat(PresetControlMetric.DELETE_PRESET.getName()), DeviceConstant.DELETE, DeviceConstant.DELETING));
+			addAdvanceControlProperties(advancedControllableProperties, createButton(stats, groupName.concat(PresetControlMetric.APPLY_PRESET.getName()), DeviceConstant.APPLY, DeviceConstant.APPLYING));
+			addAdvanceControlProperties(advancedControllableProperties, createButton(stats, groupName.concat(PresetControlMetric.SET_PRESET.getName()), DeviceConstant.SET, DeviceConstant.SETTING));
+		} else {
+			Set<String> unusedKeys = new HashSet<>();
+			unusedKeys.add(groupName.concat(PresetControlMetric.DELETE_PRESET.getName()));
+			unusedKeys.add(groupName.concat(PresetControlMetric.APPLY_PRESET.getName()));
+			unusedKeys.add(groupName.concat(PresetControlMetric.SET_PRESET.getName()));
+			removeUnusedStatsAndControls(stats, advancedControllableProperties, unusedKeys);
+		}
+		stats.put(groupName.concat(PresetControlMetric.LAST_PRESET.getName()), PresetControlMetric.PRESET.getName() + String.format(DeviceConstant.THREE_NUMBER_FORMAT,
+				Integer.parseInt(getDefaultValueForNullData(cachedLiveCameraInfo.getLastPreset(), String.valueOf(DeviceConstant.DEFAULT_VALUE)))));
+	}
+
+	/**
+	 * This method is used to handle preset control
+	 *
+	 * @param stats store all statistics
+	 * @param advancedControllableProperties store all controllable properties
+	 * @param controllableProperty controllable property
+	 * @param value value of controllable property
+	 */
+	private void presetControl(Map<String, String> stats, List<AdvancedControllableProperty> advancedControllableProperties, String controllableProperty, String value) throws
```

Comment:
javadoc does not include information about the Exception in method signature

Reply:
I fixed it